libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -254,15 +255,15 @@ class MapboxTripSession(
 
     private fun updateRawLocation(rawLocation: Location) {
         locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
+        lastLocationUpdateTimeMillis = Date().time
+        statusUpdateTimer.stopJobs()
         ioJobController.scope.launch {
             navigator.updateLocation(rawLocation)
-            lastLocationUpdateTimeMillis = Date().time
-            statusUpdateTimer.stopJobs()
-        }
-        fireOffStatusPolling()
-        statusUpdateTimer.startTimer {
-            if (Date().time - lastLocationUpdateTimeMillis >= STATUS_POLLING_INTERVAL) {
-                fireOffStatusPolling()
+            fireOffStatusPolling()
+            statusUpdateTimer.startTimer {
+                if (Date().time - lastLocationUpdateTimeMillis >= STATUS_POLLING_INTERVAL) {
```

Comment:
This check is not necessarily needed because the `MapboxTimer` will first wait the time interval and only then execute. This means, that if we're setting the interval for 1s as well, this check will always be true.

Reply:
But what in case if we will change this values? I'm agree that it is will always true for case when we have 1s for this check and 1s for timer... but if we change one of this values?